src/Assert/ZddMessageAsserter.php

```diff
@@ -2,26 +2,43 @@
 
 namespace Yousign\ZddMessageBundle\Assert;
 
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
 use Yousign\ZddMessageBundle\Factory\Property;
 use Yousign\ZddMessageBundle\Factory\PropertyList;
 
 /**
  * @internal
  */
-final class ZddMessageAssert
+final class ZddMessageAsserter
 {
+    private ?SerializerInterface $messengerSerializer;
+
+    public function __construct(SerializerInterface $messengerSerialize = null)
+    {
+        $this->messengerSerializer = $messengerSerialize;
+    }
+
     /**
      * @param class-string<object> $messageFqcn
      */
-    public static function assert(
+    public function assert(
         string $messageFqcn,
         string $serializedMessage,
         PropertyList $propertyList
     ): void {
         // ✅ Assert message is unserializable
-        $object = unserialize($serializedMessage);
+        if ($this->messengerSerializer) {
+            $encodedEnvelope = \json_decode($serializedMessage, true, 512, JSON_THROW_ON_ERROR);
```

Comment:
I think we should always use the same interface for encoding and decoding.

We cannot assess the encoding and decoding format like this.

Reply:
You're right, better now with a dedicated interface to have only one way to do the serialization in the code.